pkg/kubectl/cmd/set/set_image.go

```diff
@@ -300,6 +281,39 @@ func (o *SetImageOptions) Run() error {
 	return utilerrors.NewAggregate(allErrs)
 }
 
+func setImage(o *SetImageOptions, containers []v1.Container, containerName string, image string) (bool, bool, []error) {
+	var (
+		containerFound bool
+		transformed    bool
+		err            error
+		resolved       string
+		allErrs        []error
+	)
+	// Find the container to update, and update its image
+	for i, c := range containers {
+		if c.Name == containerName || containerName == "*" {
+			containerFound = true
+			if len(resolved) == 0 {
+				if resolved, err = o.ResolveImage(image); err != nil {
```

Comment:
It looks like this part could be done much earlier in the process (when parsing the arguments?) which would simplify this error signature since you wouldn't have to return the error.

Reply:
I looked at the code for ResolveImage, it just seems to make things more complicated than they have to be.